hledger-web/Hledger/Web/Foundation.hs

```diff
@@ -100,9 +102,11 @@ type Form x = Html -> MForm (HandlerT App IO) (FormResult x, Widget)
 instance Yesod App where
   approot = ApprootMaster $ appRoot . settings
 
-  makeSessionBackend _ =
+  makeSessionBackend _ = do
+    hledgerdata <- getXdgDirectory XdgData "hledger"
+    createDirectoryIfMissing True hledgerdata
     let sessionexpirysecs = 120
-    in  Just <$> defaultClientSessionBackend sessionexpirysecs ".hledger-web_client_session_key.aes"
+    Just <$> defaultClientSessionBackend sessionexpirysecs (hledgerdata </> "hledger-web_client_session_key.aes")
```

Comment:
I was wondering if we should add a process ID to that filename but it sounds unnecessary (#1228: *"It won't make it so the sessions will get messed up or anything like that, it's just the service (single or multiple instances) will use the same key to decrypt session for all instances, and that's ok... unless you have some unusual scenario going on"*)

Reply:
OK.